ci/notebooks-digest-updater.sh

```diff
@@ -0,0 +1,115 @@
+        # Handling PyTorch to match with regex pattern
+        if [[ $src_tag == jupyter-pytorch* ]]; then
+           src_tag="cuda-$src_tag"
+        fi
+        # This should match like for ex: jupyter-minimal-ubi9-python-3.11-20250310-60b6ecc tag name as is on quay.io registry
+        regex="^$src_tag-[0-9]{8}-$HASH$"
+        latest_tag=$(echo "${skopeo_metadata}" | jq -r --arg regex "$regex" '.RepoTags | map(select(. | test($regex))) | .[0]')
+        # use `--no-tags` for skopeo once available in newer version
+        digest=$(skopeo inspect --retry-times 3 "docker://${registry}:${latest_tag}" | jq .Digest | tr -d '"')
+        output="${registry}@${digest}"
+        echo "NEW: ${output}"
+        sed -i "s|${image}=.*|${image}=${output}|" "${PARAMS_ENV_PATH}"
+    done
+
+    update_commits "$REPO_ROOT" "$HASH"
+
+# In case the digest updater function is triggered downstream.
+elif [[ "$REPO_OWNER" == "red-hat-data-services" ]]; then
+
+    echo "This is red-hat-data-services org"
+    fetch_latest_hash "$REPO_OWNER"
+    # Get the complete list of images to update
+    IMAGES=$(grep "\-n=" "${PARAMS_ENV_PATH}" | cut -d "=" -f 1)
+    # The order of the regexes array should match with the params.env file
+    REGEXES=("^v3-[0-9]{8}-$HASH$" \
+              "^cuda-[a-z]+-minimal-[a-z0-9]+-[a-z]+-3.11-[0-9]{8}-$HASH$" \
+              "^v3-[0-9]{8}-$HASH$" \
+              "^v3-[0-9]{8}-$HASH$" \
+              "^cuda-[a-z]+-tensorflow-[a-z0-9]+-[a-z]+-3.11-[0-9]{8}-$HASH$" \
+              "^v3-[0-9]{8}-$HASH$" \
+              "^codeserver-[a-z0-9]+-[a-z]+-3.11-[0-9]{8}-$HASH$" \
+              "^rocm-[a-z]+-minimal-[a-z0-9]+-[a-z]+-3.11-[0-9]{8}-$HASH$" \
+              "^rocm-[a-z]+-pytorch-[a-z0-9]+-[a-z]+-3.11-[0-9]{8}-$HASH$" \
+              "^rocm-[a-z]+-tensorflow-[a-z0-9]+-[a-z]+-3.11-[0-9]{8}-$HASH$")
+    i=0
+    for image in ${IMAGES}; do
+        echo "CHECKING: '${image}'"
+        img=$(grep -E "${image}=" "${PARAMS_ENV_PATH}" | cut -d '=' -f2)
+        registry=$(echo "${img}" | cut -d '@' -f1)
+
+        regex=${REGEXES[$i]}
+        skopeo_metadata=$(skopeo inspect --retry-times 3 "docker://${img}")
+        latest_tag=$(echo "${skopeo_metadata}" | jq -r --arg regex "$regex" '.RepoTags | map(select(. | test($regex))) | .[0]')
+        echo "CHECKING: '${latest_tag}'"
+
+        digest=$(skopeo inspect --retry-times 3 "docker://${registry}:${latest_tag}" | jq .Digest | tr -d '"')
+        output="${registry}@${digest}"
+        echo "NEW: ${output}"
+        sed -i "s|${image}=.*|${image}=${output}|" "${PARAMS_ENV_PATH}"
+        i=$((i+1))
+    done
+
+    update_commits "$REPO_ROOT" "$HASH"
+
+else
+    echo "This script runs exclusively for the 'opendatahub-io' and 'red-hat-datascience' organizations, as it verifies/updates their corresponding quay.io registries."
```

Comment:
maybe we should end with non-zero return code in this case?

Reply:
Thanks for the review! 
Added `exit 1` on these `else`